youtube_dl/extractor/soundcloud.py

```diff
@@ -293,6 +293,11 @@ def _download_json(self, *args, **kwargs):
         if non_fatal:
```

Comment:
The condition above is too specific. The definition of `InfoExtractor._download_json()` that is being overridden makes `fatal` just a truthy/falsey value withe default `True`. It should say:
```py
        non_fatal = not kwargs.get('fatal', True)
```